sched/sched/sched_setaffinity.c

```diff
@@ -76,7 +76,7 @@ int nxsched_set_affinity(pid_t pid, size_t cpusetsize,
   irqstate_t flags;
   int ret = OK;
 
-  DEBUGASSERT(cpusetsize == sizeof(cpu_set_t) && mask != NULL);
+  DEBUGASSERT(cpusetsize == sizeof(cpu_set_t) && mask != NULL && *mask != 0);
```

Comment:
We should know that this API can be called from user code, and incorrect parameters should return corresponding return values. If  assert here directly, developers will get bad user experience:
https://man7.org/linux/man-pages/man3/pthread_setaffinity_np.3.html

```
       EINVAL (pthread_setaffinity_np()) The affinity bit mask mask
              contains no processors that are currently physically on
              the system and permitted to the thread according to any
              restrictions that may be imposed by the "cpuset" mechanism
              described in [cpuset(7)](https://man7.org/linux/man-pages/man7/cpuset.7.html).

       EINVAL (pthread_setaffinity_np()) cpuset specified a CPU that was
              outside the set supported by the kernel.  (The kernel
              configuration option CONFIG_NR_CPUS defines the range of
              the set supported by the kernel data type used to
              represent CPU sets.)

       EINVAL (pthread_getaffinity_np()) cpusetsize is smaller than the
              size of the affinity mask used by the kernel.
```